Bug 1814338 - Improve l10n GitHub Workflows

flodolo opened the pull request:
CC @gabrielluong 

This pull request restores the original cronjob time for the "import l10n" workflow, and improve the commit message for clarity.

Changes to the sync workflows are mostly for clarity:
* Use of env variables to avoid repetitions, and simplify reuse for other products (e.g. Fenix)
* Use latest versions of ubuntu and actions (most use a deprecated version of node at this point)

There should be no practical changes to the workflow, I've only used `uplift` instead of `sync` in the branch name. Both actions have been tested on my fork.

### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.





### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1814338

flodolo commented:
> The comment message should look like:

Is this supposed to say **commit** message instead? 🤔

gabrielluong commented on .github/workflows/focus-sync-strings.yml:
```diff
@@ -2,45 +2,50 @@
+      - name: Store ENV variables
+        id: env-variables
+        run: |
+          echo "project_name=Focus Strings" >> $GITHUB_OUTPUT
+          echo "beta_branch=releases_v${{ steps.discover-beta-version.outputs.beta_version }}" >> $GITHUB_OUTPUT
+          echo "branch_name=automation/uplift-focus-strings-${{ steps.discover-beta-version.outputs.beta_version }}" >> $GITHUB_OUTPUT
```
I would consider dropping `project_name` and `branch_name` as variables from these changes. I think it adds unnecessary complexity to something rather simple since these variables are only ever used once and are pretty simple strings. `beta_branch` is at least used multiple times so that is fine.

flodolo replied:
I personally don't think it adds complexity, and it puts all aspects specifics to a product in one place.
To adapt this to Fenix you would only need to update this section, instead of 2, and ending up with different versions in each place.

With that said, you're likely the one who needs to maintain these going forward, so I'll go with your preference.

gabrielluong commented on .github/workflows/focus-sync-strings.yml:
```diff
@@ -2,45 +2,50 @@
-          ref: main
-      - name: "Checkout Beta Branch"
-        uses: actions/checkout@v2
+          path: default
```
Could we just stick with doing what this was doing before with the exact path and ref? I think that we have more clarity when we are explicitly saying we are checking out the main branch compare to default branch because that begs the question of what is a  default branch and how does the actions/checkout parameters work. 

At the end of the day, I will defer to your judgement since I think the l10n team should ideally just own these actions. 

We will have to start moving away from GitHub actions to utilize taskcluster around March as preparation for the m-c migration so the amount of effort we should spend here should be fairly minimal. My concern here is that we are losing clarity about what we are doing here with this change.

flodolo commented:
Updated, rebased and pushed. Also tested on my fork.

I left `${{ github.ref_name }}` in the commit messages, since I think it's useful to make sure we're actually on the right branch.
